liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/ServerConfigDocument.java

```diff
@@ -258,5 +326,105 @@ private static boolean isValidURL(String url) {
+
+        String resolved = nodeValue;
+        Pattern varNamePattern = Pattern.compile(VARIABLE_NAME_PATTERN);
+        Matcher varNameMatcher = varNamePattern.matcher(nodeValue);
+        
+        while (varNameMatcher.find()) {
+            String variable = getProperties().getProperty(varNameMatcher.group(1));
+            
+            if (variable != null && !variable.isEmpty()) {
+                resolved = resolved.replaceAll("\\$\\{" +  varNameMatcher.group(1) + "\\}", variable);
+            }
+        }
+        return resolved;
+    }
+    
+    private static void parseVariables(Document doc) throws Exception {
+        // parse input document
+        XPath xPath = XPathFactory.newInstance().newXPath();
+        NodeList nodeList = (NodeList) xPath.compile("/server/variable").evaluate(doc, XPathConstants.NODESET);
+        
+        for (int i = 0; i < nodeList.getLength(); i++) {
+            String varName = nodeList.item(i).getAttributes().getNamedItem("name").getNodeValue();
+            String varValue = nodeList.item(i).getAttributes().getNamedItem("value").getNodeValue();
+            
+            // add unique values only
+            if (!varName.isEmpty() && !varValue.isEmpty()) {
+                props.put(varName, varValue);
+            }
+        }
+    }
+    
+    private static void parseIncludeVariables(Document doc) throws Exception {
+        // parse include document in source server xml
+        XPath xPath = XPathFactory.newInstance().newXPath();
+        NodeList nodeList = (NodeList) xPath.compile("/server/include").evaluate(doc, XPathConstants.NODESET);
+       
+        for (int i = 0; i < nodeList.getLength(); i++) {
+            String nodeValue = nodeList.item(i).getAttributes().getNamedItem("location").getNodeValue();
+            
+            if (!nodeValue.isEmpty()) {
+                Document docIncl = getIncludeDoc(nodeValue);
+                
+                if (docIncl != null) {
+                    parseVariables(docIncl);
+                    // handle nested include elements
+                    parseIncludeVariables(docIncl);
+                }
+            }
+        }
+    }
+    
+    private static void parseConfigDropinsDirVariables() throws Exception {
+        File configDropins = null;
+        
+        // if configDirectory exists and contains configDropins directory, 
+        // its configDropins has higher precedence.
+        if (configDirectory != null && configDirectory.exists()) {
+            configDropins = new File(configDirectory, "configDropins");
+        }
+        
+        if (configDropins == null || !configDropins.exists()) {
+            configDropins = new File(getServerFile().getParent(), "configDropins");
+        }
+        
+        if (configDropins != null && configDropins.exists()) {
+            File overrides = new File(configDropins, "overrides");
+            
+            if (overrides.exists()) {
```

Comment:
Shouldn't you process default first then overrides, in case the same variable is defined in both places?